packages/gatsby/src/redux/run-sift.js

```diff
@@ -123,9 +123,24 @@ function handleMany(siftArgs, nodes, sort, resolvedFields) {
  *   if `firstOnly` is true
  */
 const runSift = (args: Object) => {
-  const { getNode, addResolvedNodes } = require(`./nodes`)
+  const { getNode, addResolvedNodes, getResolvedNode } = require(`./nodes`)
 
   const { nodeTypeNames } = args
+  if (
+    args.queryArgs?.filter &&
+    Object.getOwnPropertyNames(args.queryArgs.filter).length === 1 &&
+    typeof args.queryArgs.filter?.id?.eq === `string`
+  ) {
+    // The args have an id.eq which subsumes all other queries
+    // Since the id of every node is unique there can only ever be one node found this way. Find it and return it.
+    let id = args.queryArgs.filter.id.eq
+    let node = undefined
```

Comment:
I believe you're looking for the `find` function instead of `some`.
```suggestion
    let node =  nodeTypeNames.find(typeName => {
      return !!getResolvedNode(typeName, id)
    })
```

Reply:
Um. Yeah :)

Reply:
But `find` will return typeName of the node, not the node itself in this case? Agree that `some` looks weird here though.

Reply:
Find works like filter but stops at the first one it hits.
> The find() method returns the value of the first element in the provided array that satisfies the provided testing function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

https://repl.it/@wardpeet/find-example

Reply:
Ah right. So actually, no :)

I don't want to have the `typeName` but the actual node that gets returned by `getResolvedNode` while also shortcircuiting when this is found. There's currently no array api for this, which is why I went for `.some`.

If you use `.find`, you would get the `typeName` back.

I want something like a find where the call returns the first non-undefined value that gets returned by the callback, and then stops further iteration.